x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elser/ElserMlNodeServiceSettings.java

```diff
@@ -61,21 +70,36 @@ public static ElserMlNodeServiceSettings fromMap(Map<String, Object> map) {
             validationException.addValidationError(mustBeAPositiveNumberError(NUM_THREADS, numThreads));
         }
 
+        String version = MapParsingUtils.removeAsType(map, MODEL_VERSION, String.class);
+        if (version != null) {
+            if (VALID_ELSER_MODELS.contains(version) == false) {
+                validationException.addValidationError("unknown ELSER model version [" + version + "]");
+            }
+        } else {
+            version = ElserMlNodeService.ELSER_V2_MODEL;
```

Comment:
I think this needs to choose a cleverer default. Otherwise this API will be out-of-sync with what we're doing in the UI with defaulting to the most efficient model.

The logic should be:

1. Use `ELSER_V2_MODEL_LINUX_X86` if a set containing the single item `linux-x86_64` is returned by https://github.com/elastic/elasticsearch/blob/ab93bfd972f3ba65046dc823b0f9bf174a43a1e7/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/MlPlatformArchitecturesUtil.java#L34
2. Otherwise, if that set came back empty and running in Cloud use `ELSER_V2_MODEL_LINUX_X86`
3. Otherwise, use `ELSER_V2_MODEL`

That will then match the UI logic.

So it creates a need to know if we're in Cloud. We'll have to use some existing cluster setting as a heuristic for detecting this. I think `xpack.ml.use_auto_machine_memory_percent` being `true` is the best we've got.

Reply:
Done. This caused a little churn as the `xpack.ml.use_auto_machine_memory_percent` setting and `MlPlatformArchitecturesUtil` had to be moved to core to be accessible.